### PR TITLE
Don't create overlay if disableYearOverlay is true

### PR DESCRIPTION
**Pull request by fALKENdk**
If disableYearOverlay is true don't create an overlay.

### CODE REVIEW

**Comment by fALKENdk** [comment]
Needs work, this removed the "autoclose" of the calendar.

**Comment by qodesmith** [comment]
What's the intention of this PR? To create less markup in the DOM when it's not needed?

**Comment by fALKENdk** [comment]
To not render the overlay DOM elements when the overlay is disabled and never will be opened.

**Comment by fALKENdk** [comment]
Unfortunately this still won't work because of the `.qs-overlay` and `.qs-overlay-year` query selectors.

**Comment by qodesmith** [comment]
Actually, I think you'll need to change line 852 (might be a different line in your branch) which currently reads:

```javascript
const overlayShowing = !overlay.classList.contains('qs-hidden')
```

The `overlay` variable will be null now since we're not generating the html anymore..

File: src/datepicker.js [hunk 1]
```diff
@@ -568,6 +568,8 @@ function createMonth(date, instance, overlayOpen) {
 function createOverlay(instance, overlayOpen) {
   const { overlayPlaceholder, overlayButton } = instance
 
+  if (instance.disableYearOverlay) return;
```
Comment (qodesmith) [inline src/datepicker.js]: Also, you can get rid of the semicolon. I'm not using them in this project.

**Comment by qodesmith** [comment]
Let me know if you'll make the changes, otherwise I'm going to close this PR.

**Pull request closed without merging**